sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -487,6 +487,21 @@ case class JsonToStruct(schema: StructType, options: Map[String, String], child:
   override def dataType: DataType = schema
   override def children: Seq[Expression] = child :: Nil
 
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (StringType.acceptsType(child.dataType)) {
+      try {
+        JacksonUtils.verifySchema(schema)
```

Comment:
In this case, we don't have to worry about parsing mode, `mode` because `from_json` produces `null` with the default parse mode, `FAILFAST`.